### PR TITLE
WIP - trying to run Wordcount against local Flink

### PR DESCRIPTION
**Pull request by jkff**
Do not submit, work in progress.

### CODE REVIEW

File: model/fn-execution/src/main/proto/beam_provision_api.proto [hunk 1]
```diff
@@ -67,6 +67,10 @@ message ProvisionInfo {
     // (optional) Resource limits that the SDK harness worker should respect.
     // Runners may -- but are not required to -- enforce any limits provided.
     Resources resource_limits = 4;
+
+    // (required) The artifact retrieval token produced by
```
[the remark this answers is not in the document]
Comment (ryan-williams) [reply model/fn-execution/src/main/proto/beam_provision_api.proto]: cf. #5676

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ArtifactServiceStager.java [hunk 2]
```diff
@@ -94,18 +93,20 @@ private ArtifactServiceStager(Channel channel, int bufferSize) {
...
+      throws InterruptedException {
     final Map<StagedFile, CompletionStage<ArtifactMetadata>> futures = new HashMap<>();
     for (StagedFile file : files) {
       futures.put(file, MoreFutures.supplyAsync(new StagingCallable(file), executorService));
     }
     CompletionStage<StagingResult> stagingResult =
         MoreFutures.allAsList(futures.values())
             .thenApply(ignored -> new ExtractStagingResultsCallable(futures).call());
-    return stageManifest(stagingResult);
+    return stageManifest(stagingSessionToken, stagingResult);
```
[the remark this answers is not in the document]
Comment (jkff) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ArtifactServiceStager.java]: This one's on @axelmagn

File: sdks/go/pkg/beam/util/grpcx/metadata.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-const idKey = "id"
+const idKey = "worker_id"
```
[the remark this answers is not in the document]
Comment (jkff) [reply sdks/go/pkg/beam/util/grpcx/metadata.go]: This is now fixed in the next commit.

File: sdks/go/BUILD.md [hunk 4]
```diff
@@ -61,3 +61,6 @@ to match the package structure expected by the code imports. Go users can just
 go get github.com/apache/beam/sdks/go/...
 ```
 Developers must invoke Go for cross-compilation manually, if desired.
+
+If you make changes to .proto files, you will need to rebuild the generated code.
```
[the remark this answers is not in the document]
Comment (jkff) [reply sdks/go/BUILD.md]: This is also in #5676

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ArtifactServiceStager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkff) [reply runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ArtifactServiceStager.java]: Done.

File: runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactRetrievalService.java [hunk 5]
```diff
@@ -106,8 +107,9 @@ public void getArtifact(
               .findFirst()
               .orElseThrow(
                   () ->
-                      new FileNotFoundException(
-                          String.format("Artifact location not found in manifest: %s", name)));
+                      new StatusRuntimeException(
```
[the remark this answers is not in the document]
Comment (jkff) [reply runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/artifact/BeamFileSystemArtifactRetrievalService.java]: Actually done too, as part of Axel's PR.

**Comment by jkff** [comment]
Closing this since Ankur's branch is ahead.

**Pull request closed without merging**